### PR TITLE
feat: medusa-telemetry

### PR DESCRIPTION
**Pull request by srindom**
**What**
Telemetry library for collecting anonymous usage data. 

**Why**
To understand how Medusa is being used and to get a foundation for possible improvements.

**How**
The telemetry package is adapted from gatsby-telemetry.

### CODE REVIEW

File: packages/medusa-cli/src/reporter/index.js [hunk 1]
```diff
@@ -83,24 +90,34 @@ export class Reporter {
...
    */
-  activity = message => {
+  activity = (message, config = {}) => {
+    if (config.code) {
+      track("activity_started", {
+        code: config.code,
+        message,
+      })
+    }
+
```
Comment (srindom) [inline packages/medusa-cli/src/reporter/index.js:100]: Delete

File: packages/medusa-cli/src/reporter/index.js [hunk 2]
```diff
@@ -129,6 +146,13 @@ export class Reporter {
         toLog.activity_id = activityId
         this.loggerInstance_.log(toLog)
       }
+
+      if (activity.config && activity.config.code) {
+        track("activity_progressed", {
+          code: activity.config.code,
+          message,
+        })
+      }
```
Comment (srindom) [inline packages/medusa-cli/src/reporter/index.js:155]: rm

File: packages/medusa-cli/src/reporter/index.js [hunk 3]
```diff
@@ -183,9 +208,27 @@ export class Reporter {
...
       }
+
+      if (activity.config && activity.config.code) {
+        track("activity_failed", {
+          code: activity.config.code,
+          duration: time - activity.start,
+          message,
+        })
+      }
     } else {
```
Comment (srindom) [inline packages/medusa-cli/src/reporter/index.js:219]: rm

File: packages/medusa-cli/src/reporter/index.js [hunk 4]
```diff
@@ -211,9 +255,27 @@ export class Reporter {
...
       }
+
+      if (activity.config && activity.config.code) {
+        track("activity_succeeded", {
+          code: activity.config.code,
+          duration: time - activity.start,
+          message,
+        })
+      }
     } else {
```
Comment (srindom) [inline packages/medusa-cli/src/reporter/index.js:266]: rm

**Review by olivermrbl — APPROVED** [review]
Such a nice feature!

**Pull request merged**